### PR TITLE
Imran edit summary submit requirement others

### PR DESCRIPTION
**Pull request by ImzIssa**
# Description
Gives the user permission to change the requirement to the user to submit a summary.

## Related PRS (if any):
This frontend PR is related to the [PR#942](https://github.com/OneCommunityGlobal/HGNRest/pull/942) backend PR.

## Main changes explained:
- added updateSummaryRequirements to permissionList
- added can edit check on WeeklySummaryOptions in VolunteeringTimeTab.jsx

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Follow steps in the video below

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/616a7892-7dc2-4e5d-b3d7-ae1df1d7b514

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

**Review by hbodgal — CHANGES_REQUESTED** [review]
I tried to give permission to volunteer but was not able to do it.  Let me know if I missed something. Also, if you think I did not understand the steps correctly then please provide step-b-step description to follow.
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/145157488/7e3df1fd-bcdf-40d2-b965-c9cf2baaae01

**Comment by ImzIssa** [comment]
@hbodgal The User should already have the "Edit User Profile" Permission  because a user can have "Edit User Profile" Permission and not be able to change weekly summary requirement if they don't have the "Edit Summary Submit Requirement " Permission. 
Also a user can not be able to change the weekly summary requirement if they only have "Edit Summary Submit Requirement" Permission but didn't have the "Edit User Profile" Permission in the first place.

**Review by Sandhya1236 — DISMISSED** [review]
Everything works as intended. I was able to change weekly summary options using the owner account. However, I could not make these changes with the volunteer account. After granting specific permissions—such as seeing all users, editing user profiles, changing user status, and editing summary submission requirements—from the owner account to the volunteer account, the volunteer account was updated and could change the weekly summary options.

Have a look at the video attached.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/79351059/3ea8dd87-128b-4efe-b566-53ec2d5de27c

**Comment by nishithaShetty** [comment]
Hi Imran, Nice work!
The functionality works as expected. I used admin and owner accounts to give "Edit User Profile" and "Edit Summary Submit Requirement (Others)" permissions to different accounts. I was able to see the option when the permissions were given and it was hidden when the permissions were not given. I tested this for admin, owner, volunteer, and mentor accounts.
PR#2242_1
PR#2242_2

**Review by Parth-tech — DISMISSED** [review]
Hi Imran,
The PR works as intended. The codebase is also concise and correct. Great work!

Video Reference:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56110372/6e1e3575-0cb9-44c1-8d89-b57d29f9dffc

**Comment by Parth-tech** [comment]
Hey Imran, 
I had a small question, the user who has the `updateSummaryRequirements` permission is also able to change the weekly summary option for themselves. So is this also intended or you plan to get rid of this?

**Review by Infinite-Turkey — DISMISSED** [review]
I left a comment at backend side
[review](https://github.com/OneCommunityGlobal/HGNRest/pull/942#pullrequestreview-2089961127)

**Review by CloudyZ524 — COMMENTED** [review]
Hi Imran, 
I've tested the pull request and I can successfully changed the Weekly Summary Options and Weekly Committed Hours with permission. However, I got this notifications after I press "Save Changes". Is that part of the design?
Screenshot 2024-05-31 at 22 00 56

**Review by JatinAgrawal94 — DISMISSED** [review]
1) Checked In to current branch
2) Did npm install and npm run start:local  to run this branch locally
3) Cleared site cache/data.
4) Launched app on my machine.
5) Logged in as volunteer. Wasn't authorized to change the requirement to submit a summary.
6) Logged in as admin. Made changes to user permissions to allow the user for submitting a summary.
7) Volunteer user was able to the requirements for submitting a summary.
Good Work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47699187/acb42760-9ced-4486-af57-9276f45b716c

**Review by peter6866 — DISMISSED** [review]
Great work on editing the updateSummaryRequirements permission.
When logging in as a volunteer, it's able to edit the submit summary requirement.

https://www.loom.com/share/a664305ab2784052be976cbdbc36ee50?sid=4156a36c-187b-47d5-9135-5b8b5034bd97

**Review by Kavil-Jain-514 — DISMISSED** [review]
The functionality works as expected.
The user can edit the weekly summary options after the permissions have been set for the user.
image
image

**Comment by akshit0211** [comment]
Hi Imran, it is working as you have shown in your video. Great work!

without permission
permission
with permission

**Review by akshit0211 — DISMISSED** [review]

**Review by jinxyou — DISMISSED** [review]
I was able to test the function as expected. The requirement will become editable if permission of editing were turned on, otherwise not. Great work!
Screenshot 2024-06-08 215627
Screenshot 2024-06-08 215735

File: src/components/UserProfile/UserProfile.jsx [hunk 1]
```diff
@@ -702,6 +702,7 @@ function UserProfile(props) {
   const canUpdatePassword = props.hasPermission('updatePassword');
   const canGetProjectMembers = props.hasPermission('getProjectMembers');
   const canChangeRehireableStatus = props.hasPermission('changeUserRehireableStatus');
+  const canUpdateSummaryRequirements = props.hasPermission('updateSummaryRequirements');;
```
Comment (TXMO-dev) [inline src/components/UserProfile/UserProfile.jsx:705]: Minor change on the two semicolons, it is likely to break the code.

**Review by TXMO-dev — CHANGES_REQUESTED** [review]
Please have a look at src/components/UserProfile/UserProfile.jsx on line 705, the two semicolons is likely to break the code.

File: src/components/UserProfile/UserProfile.jsx (hunk 1 above)
Comment (ImzIssa) [reply src/components/UserProfile/UserProfile.jsx]: Fixed that. Thanks.

**Review by shradac — APPROVED** [review]
When I used the owner account, I had no problems changing weekly summary options. However, when I used a volunteer account, I couldn’t make the same changes. I solved this by granting the volunteer account specific permissions from the owner account. I gave it the power to:

- See all users
- Edit user profiles
- Change user status
- Edit summary submission requirements

After I granted the volunteer account these permissions, it was able to make the changes I wanted.
Screenshot 2024-06-17 at 10 38 57 PM
Screenshot 2024-06-17 at 10 40 21 PM
Screenshot 2024-06-17 at 10 40 52 PM
Screenshot 2024-06-17 at 10 41 12 PM
Screenshot 2024-06-17 at 10 45 12 PM

**Review by DiyaWadhwani — APPROVED** [review]
Hey Imran, functionality is working as expected. I've dropped a detailed comment on the backend PR.
Thanks

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**